src/extension.ts

```diff
@@ -87,16 +97,53 @@ export async function activate(context: ExtensionContext) {
     .catch(err => {
       const message =
         "Unable to find a Java 8 or Java 11 installation on this computer. " +
-        "To fix this problem, update the 'Java Home' setting to point to a Java 8 or Java 11 home directory";
+        "To fix this problem, update the 'Java Home' setting to point to a Java 8 or Java 11 home directory " +
+        "or select a version to install automatically (window will need to be reloaded afterwards)";
+
       outputChannel.appendLine(message);
       outputChannel.appendLine(err);
-      window.showErrorMessage(message, openSettingsAction).then(choice => {
-        if (choice === openSettingsAction) {
+
+      const jabba_ver = defaultJabbaVer;
+      showJabbaMessage(message, jabba_ver);
+    });
+  commands.executeCommand("setContext", "metals:enabled", true);
+}
+
+function showJabbaMessage(
+  message: string,
+  jabba_ver: string = defaultJabbaVer
+) {
+  return window
+    .showErrorMessage(message, openSettingsAction, jdk8Action, jdk11Action)
+    .then(choice => {
+      const jdkinstaller = new JdkInstaller(outputChannel);
+      switch (choice) {
+        case openSettingsAction: {
           commands.executeCommand("workbench.action.openSettings");
+          break;
         }
-      });
+        case jdk8Action: {
+          window.withProgress({
+            location: ProgressLocation.Notification,
+            title: `Installing JDK 8, please wait...`,
+            cancellable: true
+          }, (progress, _) => {
+            return jdkinstaller.installJava(jdk8Jabba, jabba_ver).then(updateJavaConfig);
+          })
+          break;
+        }
+        case jdk11Action: {
+          window.withProgress({
+            location: ProgressLocation.Notification,
+            title: `Installing JDK 11, please wait...`,
+            cancellable: true
+          }, (progress, _) => {
```

Comment:
nit: `progress` is unused

```suggestion
          }, () => {
```